### PR TITLE
Added panning enum, enforced types and updated docs

### PR DESCRIPTION
**Pull request by KingDarBoja**
**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [X] Feature
- [X] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
See #193  

**What is the new behavior?**
Can limit panning to horizontal or vertical axis besides the default behaviour.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts [hunk 1]
```diff
@@ -912,8 +913,8 @@ export class GraphComponent extends BaseChartComponent implements OnInit, OnChan
    *
    */
   @HostListener('document:touchmove', ['$event'])
-  onTouchMove($event: any): void {
-    if (this.isPanning && this.panningEnabled) {
+  onTouchMove($event: TouchEvent): void {
```
Comment (marjan-georgiev) [inline projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts]: The `TouchEvent` type causes issues in some older versions of browsers, so we have to remove it.

File: projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts [hunk 2]
```diff
@@ -311,7 +312,7 @@ export class GraphComponent extends BaseChartComponent implements OnInit, OnChan
   createGraph(): void {
     this.graphSubscription.unsubscribe();
     this.graphSubscription = new Subscription();
-    const initializeNode = n => {
+    const initializeNode = (n: Node | ClusterNode) => {
```
Comment (marjan-georgiev) [inline projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts]: ClusterNode extends from Node, so `Node` would be enough here.

**Comment by KingDarBoja** [comment]
Roger that! I will make it as soon as I get time :+1:

**Comment by marjan-georgiev** [comment]
Looks good! Thanks @KingDarBoja

**Pull request merged**